src/deploy/functions/backend.ts

```diff
@@ -0,0 +1,356 @@
+import * as proto from "../../gcp/proto";
+import * as gcf from "../../gcp/cloudfunctions";
+import * as cloudscheduler from "../../gcp/cloudscheduler";
+import * as utils from "../../utils";
+import { FirebaseError } from "../../error";
+import { Context } from "./args";
+
+export interface ScheduleRetryConfig {
+  retryCount?: number;
+  maxRetryDuration?: proto.Duration;
+  minBackoffDuration?: proto.Duration;
+  maxBackoffDuration?: proto.Duration;
+  maxDoublings?: number;
+}
+
+export interface PubSubSpec {
+  id: string;
+  project: string;
```

Comment:
Not sure how I feel about having `project` show up in each of these specs. AFAIK we aren't contemplating cross-project resource creation, and a given deploy only ever affects a single project. Do we need to embed it everywhere?